client/hijack.go

```diff
@@ -72,7 +74,21 @@ func (cli *Client) postHijacked(ctx context.Context, path string, query url.Valu
 	defer clientconn.Close()
 
 	// Server hijacks the connection, error 'connection closed' expected
-	_, err = clientconn.Do(req)
+	resp, err := clientconn.Do(req)
+	if err != nil {
+		return types.HijackedResponse{}, err
+	}
+	if resp != nil {
+		if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusSwitchingProtocols {
+			defer resp.Body.Close()
+
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return types.HijackedResponse{}, err
+			}
+			return types.HijackedResponse{}, fmt.Errorf("Error response from daemon: %s", bytes.TrimSpace(body))
+		}
+	}
```

Comment:
@dnephin ，thanks for your review and advice. 
I just want the response close called when resp.StatusCode != http.StatusOK(or http.StatusSwitchingProtocols)， If  resp.StatusCode is http.StatusSwitchingProtocols, do we must close the resp ? I'm not sure.